### DISCUSSION TITLE
Icon Request: icon-nuclear-mushroom

### DISCUSSION THREAD

**Discussion by ozzargueyo** [FortAwesome/Font-Awesome#21229] [Icon Requests]
Hi, i would like to have an icon for a nuclear mushroom, I now its a weird request but i could be useful for some applications.

Thanks and greetings from Argentina.

**Comment by CubicWebs** [comment]
Not sure what one is but sounds cool :) +1

**Comment by brodul** [comment]
+1 Especial useful for font encoding fixes.

**Comment by Oceanswave** [comment]
+1 for this and including an example

![image](https://cloud.githubusercontent.com/assets/760674/6052877/2ec30494-aca6-11e4-8101-0943fe653926.png)

"Occasionally, we find ourselves wanting to implement a button or indicating an activity that describes a very drastic option.
Often, we refer to this as "The Nuclear Option" or possibly "Nuke it"

Thus, it would be wonderful, just stellar, I mean completely radical, if font awesome had an icon to support such functionality -- e.g. a mushroom cloud icon"

A live, in the wild, example of the usefulness of such an icon is the shockwave porn subreddit (SFW) http://www.reddit.com/r/shockwaveporn

**Comment by jrswenson** [comment]
+1

**Comment by moebaca** [comment]
Seems to still be open as of 3 years. Any other service provide an icon similar to that linked above?

**Comment by pegarm** [comment]
+1

**Comment by threedot14** [comment]
+1
Would like this icon to use for an action that while serving a purpose, it should be taken rarely and under extreme caution. (In my case: it's a "purge/delete-permanently" action.)

**Comment by sensibleworld** [comment]
This is a quick update to let folks know that we’re refining our icon request workflow behind the scenes, and as part of that we’re going through all open requests and filing them into several new buckets to be addressed.

In this case we’ve added this issue to our new list to be created! We can’t promise any firm timelines right now, but this will help us keep track of it.

We’ll update this issue when the icon is released into the wild 🙂

**Comment by jdx** [comment]
I investigated this and found the root cause.

## How Redactions Work

The redaction patterns (like `*KEY*`) are matched against environment variable **names**, and then the **values** of those matched variables are redacted from output.

For example:
- `API_KEY=mysecret123`
- Pattern `*KEY*` matches the name `API_KEY`
- The value `mysecret123` gets redacted

## The Bug

The problem occurs when an env var matching your pattern has a **very short value** (like `1`, `0`, `true`, etc.). That short string gets redacted everywhere in output, including inside ANSI escape sequences.

For example, if you have some env var like `SOME_KEY=1`, then:
- `\x1b[1m` (ANSI bold) becomes `\x1b[[redacted]m`

## Workaround

Check which of your env vars match `*KEY*`, `*SECRET*`, `*TOKEN*`, `*PASSWORD*` and see if any have short values. You can narrow your patterns to be more specific, e.g. `*API_KEY` instead of `*KEY*`.

## Fix

I'll submit a PR to add a minimum length filter (e.g., skip values < 3 characters) to prevent short strings from being redacted, as they're unlikely to be secrets and cause these false positive issues.
